tools/apigee-envoy-quickstart/extension/README.md

```diff
@@ -0,0 +1,145 @@
+# Apigee Envoy extension (for external access)
+
+This is an extension to the starter setup of apigee-envoy deployment within GKE container platform with Apigee X/Hybrid as API management platform. 
+
+The [starter kit](https://github.com/apigee/devrel/tree/main/tools/apigee-envoy-quickstart#envoy-with-apigee-adapter-as-containers-within-kubernetes-platform) deploys httpbin service configured with Envoy proxies as side car proxies. The Envoy proxy is enabled with apigee-adapter as step in the request path enforcing  Apigee provided authentication methods.
+
+This extension enables the exposure of deployed sample application (httpbin) externally via istio-ingressgateway. It showcases Apigee enabled envoy proxies can offer protection of API traffic intiated outside the mesh. 
+
+![poc-setup](../assets/istio-apigee-envoy-external.png)
+
+## Tools
+
+* kubectl v1.23.0
+* bash 5.1.8
+* envsubst 0.21
+
+### Pre-requisities:
+
+1. Deployment of [quickstart setup](https://github.com/apigee/devrel/tree/main/tools/apigee-envoy-quickstart#envoy-with-apigee-adapter-as-containers-within-kubernetes-platform) of apigee-envoy adapter within Istio enabled Kubernetes platform. 
+
+1. Ensure Kubernetes cluster is running with istio-ingressgateway and supports external load balancers.(Execute [step 4 and 5](#set-the-namespace-hosting-istio-ingressgateway) below to setup ingressgateway, if missing within the cluster)
+
+### Installation:
+
+1. **Set environment variables.** 
+    ```bash
+    export USE_GKE_GCLOUD_AUTH_PLUGIN=True
+    export GKE_PROJECT_ID=<gcp-project-id-hosting-the-cluster>
+    export CLUSTER_NAME=<gke-cluster-name>
+    export CLUSTER_LOCATION=<gke-cluster-region>
+    export APIGEE_ORG=<apigee-org-tied-to-apigee-envoy-adapter>
+    ```
+
+1. **Set up local authentication to your project.**
+    ```bash
+    gcloud config set project $GKE_PROJECT_ID
+    gcloud auth application-default login --no-launch-browser
+
+    export TOKEN=$(gcloud auth print-access-token)
+    ```
+
+1. **Set up login profile for your cluster.**
+    ```bash
+    gcloud container clusters get-credentials "$CLUSTER_NAME" \
+     --zone "$CLUSTER_LOCATION" \
+     --project "$GKE_PROJECT_ID"
+     export CLUSTER_CTX="gke_${GKE_PROJECT_ID}_${CLUSTER_LOCATION}_${CLUSTER_NAME}"
+    ```
+
+1. ### Set the namespace hosting istio-ingressgateway
```

Comment:
How do we ensure that these instructions are compatible with the istio/asm version that a user has installed?

Reply:
which of these commands are specific to a version for istio/asm ? May be the the tool should set the expected platform whats its coded/tested for and ask users to modify it for their respective versions? As this is just a 'quickstart' tool.

Thoughts?

Reply:
I can't see any Istio or ASM specific commands, but I guess it's just a question of how compatible this tool should be. On this occassion, @ganadurai makes a valid point that this is all in the context of a highly opinionated "quickstart" which gets you up and running quickly, but potentially has limited flexibility and compatibility.

It's likely @danistrebel is actually referring to the *version* of Istio and not the distribution (e.g. ASM). I'm not sure what guarantees can be put in place for that. Perhaps the scripts could do a check of the Istio version currently installed and ensure it's at `> x.x.x` 🤔

Reply:
just a suggestion. Happy to leave it as is.